src/main/puzzle/scenario/scenario.gd

```diff
@@ -12,6 +12,7 @@ func _ready() -> void:
 	PuzzleScore.reset() # erase any lines/score from previous games
 	PuzzleScore.connect("game_ended", self, "_on_PuzzleScore_game_ended")
 	PuzzleScore.connect("after_game_prepared", self, "_on_PuzzleScore_after_game_prepared")
+	$Puzzle.get_milestone_hud().init_milebar()
```

Comment:
This warrants a comment, as the bug you are fixing with this is a weird one.

Reply:
Added detailed comment, `Intuitively MilestoneHud could initialize its own milebar on _ready...`